jlm/rvsdg/simple-node.hpp

```diff
@@ -120,6 +134,31 @@ SimpleNode::output(size_t index) const noexcept
   return static_cast<simple_output *>(Node::output(index));
 }
 
+struct CreateOpNodeResult
+{
+  Node * node;
+  std::vector<output *> outputs;
+};
+
+template<typename OpType, typename... OpParameters>
+CreateOpNodeResult
+CreateOpNode(const std::vector<output *> & arguments, OpParameters... opParameters)
+{
+  Region * region = arguments.at(0)->region();
```

Comment:
This method assumes at least a single operand. What about nullary nodes? Could we document the constraints as well as insert a `JLM_ASSERT()` at the beginning rather than implicit relying on c++ exceptions via `at()`

Reply:
added another method that allows to create nullary nodes and reworked checking